src/common/types/types.cpp

```diff
@@ -149,6 +149,22 @@ std::unique_ptr<StructField> StructField::copy() const {
     return std::make_unique<StructField>(name, type->copy());
 }
 
+StructTypeInfo::StructTypeInfo(std::vector<std::unique_ptr<StructField>> fields)
```

Comment:
Why put constructor implementation here after `copy()`? I'd like to try to align the order of function definition and implementation in both .h and .cpp file.